### ISSUE TITLE
Gulp test can not find module

### ISSUE THREAD

**Issue by kostonstyle** [Swiip/generator-gulp-angular#364]
Hi
I have following question.
http://stackoverflow.com/questions/28593992/gulp-test-can-not-find-module

Thanks very much

**Comment by Swiip** [comment]
Hi, try to activate debug logs of Karma in `karma.conf.js`. I suspect that your files are not injected into Karma, the logs will list the files...

**Comment by kostonstyle** [comment]
I am pretty newies, how can I activate debug logs of Karma in karma.conf.js?

**Comment by kostonstyle** [comment]
This my karma.conf.js content
```javascript
		module.exports = function(config) {
		  config.set({
			autoWatch : false,

			frameworks: ['jasmine'],

			browsers : ['PhantomJS'],

			plugins : [
				'karma-phantomjs-launcher',
				'karma-jasmine'
			],

			logLevel: config.LOG_DEBUG

		  });
		};
```

When I execute gulp test I've got

	`$ gulp test
	[09:37:47] Using gulpfile D:\angularjs\project\gulpfile.js
	[09:37:47] Starting 'test'...
	[09:37:48] Starting Karma server...
	DEBUG [plugin]: Loading plugin karma-phantomjs-launcher.
	DEBUG [plugin]: Loading plugin karma-jasmine.
	INFO [karma]: Karma v0.12.31 server started at http://localhost:9876/
	INFO [launcher]: Starting browser PhantomJS
	DEBUG [temp-dir]: Creating temp dir at C:\cygwin64\tmp\karma-63050734
	DEBUG [launcher]: D:\angularjs\project\node_modules\karma-phantomjs-launcher\node_modules\phantomjs\lib\phantom\bin\phantomjs C:\cygwin64\tmp\karma-63050734/capture.js
	DEBUG [launcher]: Process PhantomJS exited with code -4058
	ERROR [launcher]: Cannot start PhantomJS
			Can not find the binary D:\angularjs\project\node_modules\karma-phantomjs-launcher\node_modules\phantomjs\lib\phantom\bin\phantomjs
			Please set env variable PHANTOMJS_BIN
	DEBUG [temp-dir]: Cleaning temp dir C:\cygwin64\tmp\karma-63050734
	DEBUG [watcher]: Resolved files:
			D:/angularjs/project/node_modules/jasmine-core/lib/jasmine-core/jasmine.js
			D:/angularjs/project/node_modules/karma-jasmine/lib/boot.js
			D:/angularjs/project/node_modules/karma-jasmine/lib/adapter.js
			D:/angularjs/project/bower_components/jquery/dist/jquery.js
			D:/angularjs/project/bower_components/angular/angular.js
			D:/angularjs/project/bower_components/angular-animate/angular-animate.js
			D:/angularjs/project/bower_components/angular-sanitize/angular-sanitize.js
			D:/angularjs/project/bower_components/lodash/dist/lodash.compat.js
			D:/angularjs/project/bower_components/restangular/dist/restangular.js
			D:/angularjs/project/bower_components/angular-ui-router/release/angular-ui-router.js
			D:/angularjs/project/bower_components/bootstrap/dist/js/bootstrap.js
			D:/angularjs/project/bower_components/angular-strap/dist/angular-strap.js
			D:/angularjs/project/bower_components/angular-strap/dist/angular-strap.tpl.js
			D:/angularjs/project/bower_components/angular-mocks/angular-mocks.js
			D:/angularjs/project/src/app/app.config.js
			D:/angularjs/project/src/app/app.module.js
			D:/angularjs/project/src/app/app.routes.js
			D:/angularjs/project/src/app/main/main.controller.js
			D:/angularjs/project/src/app/main/main.controller.spec.js
			D:/angularjs/project/src/app/services/text.service.js
			D:/angularjs/project/src/app/services/text.service.spec.js
			D:/angularjs/project/src/app/signup/signup.controller.js
			D:/angularjs/project/src/components/navbar/navbar.controller.js
	DEBUG [launcher]: PhantomJS failed (cannot start). Not restarting.
	DEBUG [karma]: Run complete, exiting.
	DEBUG [launcher]: Disconnecting all browsers

	D:\angularjs\project\gulp\unit-tests.js:30
		  throw err;
				^
	Error: karma exited with code 1
		at done (D:\angularjs\project\node_modules\gulp-karma\index.js:56:30)
		at ChildProcess.<anonymous> (D:\angularjs\project\node_modules\gulp-karma\index.js:82:7)
		at ChildProcess.emit (events.js:110:17)
		at Process.ChildProcess._handle.onexit (child_process.js:1067:12)
	`
What am I doing wrong?

**Comment by kostonstyle** [comment]
Please help me. I really do not know, what to do.

**Comment by Swiip** [comment]
Did you try this suggestion, it seems interesting to me.

http://stackoverflow.com/a/28602223

**Comment by kostonstyle** [comment]
Yes I try it, but does not help.

**Comment by kostonstyle** [comment]
Here is the solution
http://stackoverflow.com/questions/28593992/gulp-test-can-not-find-module

**Comment by quipu** [comment]
I had a similar problem, caused by the files being injected in the wrong order.
By logging out the files I could see the files were being injected but not in order of dependency. If I took the log output and put that into the karma config file in the right order, then run karma directly, it worked fine.
To continue to enjoy the benefits of auto-injecting, I edited the gulp unit-tests.js file to use of angularFilesort to correctly order the files according to dependency, and all working now when running gulp test.

```
// gulp/unit-tests.js
// after line 36 inserted the following line:
.pipe($.angularFilesort())
// which is followed by .pipe(concat(function(files) {
```
